add tiflash config storage.temp (#21376)
pingcap/docs: tiflash/tiflash-configuration.md

- The path in which the TiFlash temporary files are stored.
- By default, it is the first directory in [`path`](#path) or in [`storage.latest.dir`](#dir-1) appended with `"/tmp"`.

> - Starting from v9.0.0, it is recommended that you use the [`storage.temp`](#storagetemp-new-in-v900) configuration instead of `tmp_path`, as it supports capacity limits to control temporary file space usage.

Reviewer: ```suggestion
- Starting from v9.0.0, it is recommended that you use the [`storage.temp`](#storagetemp-new-in-v900) configuration instead of `tmp_path`, because it supports setting a capacity limit to control temporary file space usage.
```

> - When the `storage.temp` configuration exists, the `tmp_path` configuration is ignored.

Reviewer: ```suggestion
- When `storage.temp` is configured, the `tmp_path` configuration does not take effect.
```

<!-- Example: `[10737418240, 10737418240]` -->

#### storage.temp <span class="version-mark">New in v9.0.0</span>

##### `dir`

> - The path in which the temporary spill files generated during queries are stored.

Reviewer: ```suggestion
- The directory in which the temporary spill files generated during query execution are stored.
```

- By default, it is the first directory in [`storage.latest.dir`](#dir-1) appended with `"/tmp"`.

##### `capacity`

- Limits the total space usage of the temporary file directory. If the temporary spill files generated during query execution exceed this limit, the query fails with an error.
- Unit: Byte. Formats such as `"10GB"` are not supported.
- Range: `[0, 9223372036854775807]`

> - If this value is not set or set to `0`, temporary files are not subject to a space limit and can use the entire disk capacity.

Reviewer: ```suggestion
- If this value is not set or is set to `0`, temporary files are not subject to a space limit and can use the entire disk capacity.
```
